lib/GridItem.jsx

```diff
@@ -171,11 +171,11 @@ export default class GridItem extends React.Component<Props, State> {
 
   currentNode: HTMLElement;
 
-  componentWillReceiveProps(nextProps: Props) {
-    const { droppingPosition } = nextProps;
+  componentDidUpdate(prevProps: Props) {
+    const { droppingPosition } = this.props;
     const { dragging } = this.state;
 
-    if (!droppingPosition || !this.props.droppingPosition) {
+    if (!droppingPosition || !prevProps.droppingPosition) {
```

Comment:
TBH this bailout here makes this code hard to read. #980 should have refactored the droppable functionality into another function so this logic is more readable.

Reply:
@STRML thank you for the comment! I'll fix it right after the performance issue